src/main/java/com/opentable/metrics/JettyServerMetricsConfiguration.java

```diff
@@ -72,9 +72,10 @@
             super(metricRegistry,
                 32, 32, // Default number of threads, overridden in otj-server EmbeddedJetty
                 30000,  // Idle timeout, irrelevant since max == min
-                new BlockingArrayQueue<>(qSize, // Initial queue size
-                    8, // Expand increment (irrelevant; initial == max)
-                    qSize)); // Upper bound on work queue
+                // NB: we originally wished to size this queue, but unfortunately Jetty leaks resources
+                // when you throw RejectedExecutionException due to a full work queue.
+                // So we leave it unbounded since even an OOME is preferable.
+                new BlockingArrayQueue<>());
```

Comment:
This is really unfortunate, but looks like it's the best compromise among available options.

We discussed in-person another idea: have a queue implementation with a limit, and when the queue size is surpassed, it terminates the process.  However, that's pretty drastic.  Really, all we want is to drop connections when we exceed the queue size.  But Jetty "doesn't support that" presently because of the rejected execution exception problem.

At least this way, the process will crash eventually instead of just running out of file descriptors.

@rushalias, didn't you have a pretty succinct reproducer?  Let's send that to the Jetty team to build on that thread where @stevenschlansker reported the issue in prose.